core/src/main/java/org/ehcache/core/config/DefaultConfiguration.java

```diff
@@ -77,7 +78,7 @@ public DefaultConfiguration(ClassLoader classLoader, ServiceCreationConfiguratio
   public DefaultConfiguration(Map<String, CacheConfiguration<?, ?>> caches, ClassLoader classLoader, ServiceCreationConfiguration<?>... services) {
     this.services = unmodifiableCollection(Arrays.asList(services));
     this.caches = new ConcurrentHashMap<String, CacheConfiguration<?, ?>>(caches);
-    this.classLoader = classLoader;
+    this.classLoader = classLoader == null ? ClassLoading.getDefaultClassLoader() : classLoader;
```

Comment:
A similar check should be made on the copy constructor, or even have it delegate to this one to simplify.